test/__runner__/browser/index.js

```diff
@@ -124,23 +118,35 @@ function generate(options) {
 
 function runTests(options) {
     return generate(options).then(generated => {
-        console.log(`Running ${generated.url} using mocha-phantomjs...`);
-        var mochaPhantomJSOptions = { useColors: true };
+        var cleanup = JSDOM(fs.readFileSync(generated.url, 'utf-8'), {
+            url: 'file://' + generated.url,
+            runScripts: 'dangerously',
+            pretendToBeVisual: true,
+            resources: 'usable'
+        });
 
-        if (shouldCover) {
-            mochaPhantomJSOptions.hooks = 'mocha-phantomjs-istanbul';
-            console.log(options.testsFile)
-            mochaPhantomJSOptions.coverageFile = getCoverageFile(options.testsFile);
-        }
+        return new Promise(function (resolve, reject) {
+            window.addEventListener('load', function () {
+                var runner = window.MOCHA_RUNNER;
+                runner.on('end', function () {
+                    if (shouldCover) {
+                        var coverageFile = getCoverageFile(options.testsFile);
+                        fs.writeFileSync(coverageFile, JSON.stringify(window.__coverage__));
```

Comment:
If `writeFileSync` throws, the tests will hang because the promise will never reject. Additionally, since this code is already async, why not just use `fs.writeFile` instead?

Reply:
Good catch. I thought of switching it but it was just left over from copy pasting.

I’ll clean that up.